src/gardenlinux/s3/s3_artifacts.py

```diff
@@ -116,36 +117,20 @@ def upload_from_directory(
 
         artifacts_dir = Path(artifacts_dir)
 
+        cname_object = CName(cname)
+
         if not artifacts_dir.is_dir():
             raise RuntimeError(f"Artifacts directory given is invalid: {artifacts_dir}")
 
         release_file = artifacts_dir.joinpath(f"{cname}.release")
         release_timestamp = stat(release_file).st_ctime
 
-        release_config = ConfigParser(allow_unnamed_section=True)
-        release_config.read(release_file)
-
-        # Get architecture from the GARDENLINUX_CNAME (second to last element when split by -)
-        release_cname = release_config.get(UNNAMED_SECTION, "GARDENLINUX_CNAME")
-        cname_parts = release_cname.split("-")
-        if len(cname_parts) < 2:
-            raise RuntimeError(f"Invalid GARDENLINUX_CNAME format in release file: {release_cname}")
-        arch = cname_parts[-2]  # Second to last element is the architecture
+        cname_object.load_from_release_file(release_file)
```

Comment:
Is the `load_from_release_file` function guaranteed to never touch the `features` directory? as in will this also work entirely without the main GL repo checked out given just the release file?